### PR TITLE
Ekw/workflow templates

### PR DESCRIPTION
**Pull request by EmilKWarmdahl**
## Purpose

Create a reusable workflow for building and publishing docker images

## Changes

Added reusable workflow

## Checklist

- [x] My code follows the style of this project.
- [x] The code compiles without warnings.
- [x] I have performed a self-review of the changes.
- [x] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [x] (If necessary) I have updated the CHANGELOG.

## CLA acceptance

_Remove if not applicable.

By submitting the contribution I accept the terms and conditions of the
Contributor License Agreement v1.0
- link: https://developers.concordium.com/CLAs/Contributor-License-Agreement-v1.0.pdf

- [x] I accept the above linked CLA.

### CODE REVIEW

File: .github/workflows/docker-release-workflow.yaml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+          exitcode=$?
+          if [ $exitcode -eq "0" ]; then
+            echo "Error: ${{ env.FULL_IMAGE_TAG }} already exist"
+            exit 1
+          fi
+
+      - name: Clone repo
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ github.ref_name }}
```
Comment (lassemand) [inline .github/workflows/docker-release-workflow.yaml:59]: This should recursively clone

File: .github/workflows/docker-release-workflow.yaml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+      - name: Build Docker Image
+        uses: docker/build-push-action@v6
+        with:
+          context: .
+          file: ${{ inputs.DOCKER_FILE }}
+          tags: ${{ env.FULL_IMAGE_TAG }}
+          no-cache: true
+          push: false
+          build-args: |
+            ${{inputs.BUILD_ARGS}}
```
Comment (lassemand) [inline .github/workflows/docker-release-workflow.yaml:71]: Have you tested this with mulitple args?

Comment (EmilKWarmdahl) [reply .github/workflows/docker-release-workflow.yaml:71]: I have tested with:

```
rust_version=1.73.0
ghc_version=9.6.4
protoc_version=3.15.3
flatbuffers_tag=v22.12.06
nvm_sh_version=v0.37.2
cmake_version=3.25.1
```

File: .github/workflows/docker-release-workflow.yaml (hunk 1 above)
Comment (EmilKWarmdahl) [reply .github/workflows/docker-release-workflow.yaml:59]: Yes, you are right.

**Review by lassemand — APPROVED** [review]

**Comment by lassemand** [comment]
We should also validate whether the version on the file system correlates with the one defined in the release tag. Just take the version as input and let the caller figure out how to obtain it

**Pull request merged**